streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollector.java

```diff
@@ -81,7 +81,9 @@ public RecordCollector(Producer<byte[], byte[]> producer) {
             if (partitions != null)
                 partition = partitioner.partition(record.key(), record.value(), partitions.size());
         }
-        this.producer.send(new ProducerRecord<>(record.topic(), partition, keyBytes, valBytes), callback);
+        this.producer.send(new ProducerRecord<>(record.topic(), partition, record.timestamp(),
```

Comment:
I meant the `org.apache.kafka.streams.state.internals.StoreChangeLogger` class, which also used a `RecordCollector` reference but does not explicitly set the timestamp.